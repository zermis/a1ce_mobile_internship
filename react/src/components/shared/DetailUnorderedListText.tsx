import React from 'react';
import { Text, StyleSheet, View } from 'react-native';
import Colors from '../../constants/Colors';
import { RelatedCompetency } from '../../models/Calendar';

interface Props {
    relatedCompetencies: RelatedCompetency[];
    style?: object;
}

const DetailUnorderedListText = (props: Props) => {
    return (
        <View>
            {props.relatedCompetencies.map(
                (competency: RelatedCompetency, index: number) => (
                    <View style={styles.unorderedListContainer} key={index}>
                        <Text style={styles.unorderedListDot}>{'\u2022'}</Text>
                        <Text style={styles.unorderedListText}>
                            {competency.title}
                        </Text>
                    </View>
                )
            )}
        </View>
    );
};

const styles = StyleSheet.create({
    unorderedListContainer: {
        flexDirection: 'row'
    },
    unorderedListDot: {
        color: Colors.textBody
    },
    unorderedListText: {
        paddingLeft: 5,
        color: Colors.textBody,
        fontFamily: 'ibm-regular',
        fontSize: 14
    }
});

export default DetailUnorderedListText;
