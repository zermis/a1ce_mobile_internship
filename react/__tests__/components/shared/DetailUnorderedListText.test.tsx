import DetailUnorderedListText from '../../../src/components/shared/DetailUnorderedListText';
import renderer from 'react-test-renderer';
import React from 'react';

it('render detail unordered list text correctly', function () {
    var tree = renderer
        .create(<DetailUnorderedListText relatedCompetencies={[]} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
//
