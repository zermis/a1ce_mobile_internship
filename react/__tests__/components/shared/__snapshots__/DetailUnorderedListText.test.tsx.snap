// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render detail unordered list text correctly 1`] = `<View />`;
